### DISCUSSION TITLE
LDM Notes for February 16th, 2022

### DISCUSSION THREAD

**Discussion by 333fred** [dotnet/csharplang#5790] [LDM Notes]
https://github.com/dotnet/csharplang/blob/main/meetings/2022/LDM-2022-02-16.md

## Agenda

1. Open questions in `field`
2. Triage
    1. User-defined positional patterns
    2. Delegate type arguments improvements
    3. Practical existential types for interfaces
    4. Static abstract interfaces and static classes

**Comment by pinkfloydx33** [comment]
The link under "Delegate type arguments improvements" is wrong (same link used in the following section)

**Reply by 333fred** [reply]
Thanks, fixed.

**Comment by DavidArno** [comment]
Regarding User-defined positional patterns:

> ... we see active patterns as one of the last pattern-related things that need to be added to the pattern feature to make it generally "complete", and are excited to look at them after we land the current pattern work ... Into the backlog, for after we finish the current pattern features.


This news makes me very happy. Bringing pattern matching to C# - being a language not designed for it initially - has been a long, hard journey for the language team. It's been a massive success so far though and in my view it's one of the languages greatest achievements in recent years. Really looking forward to it being "complete" in the next few years therefore.
